jgrapht-core/src/test/java/org/jgrapht/perf/clique/MaximalCliqueEnumerationPerformanceTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.perf.clique;
+
+import java.util.Iterator;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.jgrapht.Graph;
+import org.jgrapht.UndirectedGraph;
+import org.jgrapht.alg.clique.BronKerboschCliqueFinder;
+import org.jgrapht.alg.clique.DegeneracyBronKerboschCliqueFinder;
+import org.jgrapht.alg.clique.PivotBronKerboschCliqueFinder;
+import org.jgrapht.generate.GnpRandomGraphGenerator;
+import org.jgrapht.generate.GraphGenerator;
+import org.jgrapht.graph.DefaultEdge;
+import org.jgrapht.graph.IntegerVertexFactory;
+import org.jgrapht.graph.SimpleGraph;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+import junit.framework.TestCase;
+
+/**
+ * A small benchmark comparing maximal clique enumeration algorithms.
+ * 
+ * @author Dimitrios Michail
+ */
+public class MaximalCliqueEnumerationPerformanceTest
```

Comment:
Based on the results of this class, can you add some recommendations to the javadoc in the various maximal clique enumeration classes? Something like:
"On average, the best performance is achieved with MaxCliqueEnumerator X. If you have a sparse graph, it may be better to use MaxCliqueEnumerator Y. The best runtime complexity is achieved by MaxCliqueEnumerator Z."

I don't know whether something like that is possible. It may quite well be the case that neither of the implementations dominates all the other implementations. So this is just a suggestion.